rclcpp/test/test_publisher.cpp

```diff
@@ -80,16 +80,91 @@ TEST_F(TestPublisher, construction_and_destruction) {
   initialize();
   using rcl_interfaces::msg::IntraProcessMessage;
   {
-    auto publisher = node->create_publisher<IntraProcessMessage>("topic");
+    auto publisher = node->create_publisher<IntraProcessMessage>("topic", 42);
+    (void)publisher;
   }
 
   {
     ASSERT_THROW({
-      auto publisher = node->create_publisher<IntraProcessMessage>("invalid_topic?");
+      auto publisher = node->create_publisher<IntraProcessMessage>("invalid_topic?", 42);
     }, rclcpp::exceptions::InvalidTopicNameError);
   }
 }
 
+/*
+   Testing publisher creation signatures.
+ */
+TEST_F(TestPublisher, various_creation_signatures) {
+  initialize();
+  using rcl_interfaces::msg::IntraProcessMessage;
+  {
+    auto publisher = node->create_publisher<IntraProcessMessage>("topic", 42);
+    (void)publisher;
+  }
+  {
+    auto publisher = node->create_publisher<IntraProcessMessage>("topic", rclcpp::QoS(42));
+    (void)publisher;
+  }
+  {
+    auto publisher =
+      node->create_publisher<IntraProcessMessage>("topic", rclcpp::QoS(rclcpp::KeepLast(42)));
+    (void)publisher;
+  }
+  {
+    auto publisher =
+      node->create_publisher<IntraProcessMessage>("topic", rclcpp::QoS(rclcpp::KeepAll()));
+    (void)publisher;
+  }
+  {
+    auto publisher =
+      node->create_publisher<IntraProcessMessage>("topic", 42, rclcpp::PublisherOptions());
+    (void)publisher;
+  }
+  {
+    auto publisher =
+      rclcpp::create_publisher<IntraProcessMessage>(node, "topic", 42, rclcpp::PublisherOptions());
+    (void)publisher;
+  }
+  // Now deprecated functions.
+#if !defined(_WIN32)
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wdeprecated-declarations"
+#else  // !defined(_WIN32)
+# pragma warning(push)
+# pragma warning(disable: 4996)
+#endif
+  {
+    // auto publisher = node->create_publisher<IntraProcessMessage, std::allocator<void>, rclcpp::Publisher<IntraProcessMessage, std::allocator<void>>>("topic");
+    auto publisher = node->create_publisher<IntraProcessMessage>("topic");
+    (void)publisher;
+  }
+#if 0
```

Comment:
I suppose the code between `#if 0` and `#endif` will be cleaned.

Reply:
Same.